manuscript/13-Modules.md

```diff
@@ -456,13 +456,186 @@ This example loads `module.js` as a module instead of a script by passing a seco
+
+As with single files, packages can contain both `.mjs` and `.js` files to ensure compatibility with older versions of Node.js.
+
+A> #### Module Resolution Differences Between the Browser and Node.js
+A>
+A> The ECMAScript 6 standard does not specify the format of module filenames in `import` statements, leaving those details up to the implementations. As such, there's a subtle but important difference in the way browsers and Node.js resolve modules. Consider the following:
+A>
+A> ```js
+A> import { sum } from "example/sum.js";
+A> ```
+A>
+A> While this code may look simple to understand, browsers and Node.js interpret `"example/sum.js"` differently. The differences are outlined in the table below.
+A>
+A> | **Environment** | **File to Load** | **Loaded As** | **Mode** |
+A> -----------------------------------------------------------------
+A> | Browser         | `./example/sum.js` | JavaScript module | strict |
+A> | Node.js         | `./node_modules/example/sum.js` | CommonJS module | nonstrict |
+A>
+A> In a browser, `"example"` is resolved as a subdirectory whereas Node.js resolves `"example"` as a package name. Further, because the `.js` file extension is specified (which is required by the browser), the browser loads the file as a JavaScript module while Node.js loads the file as CommonJS. These differences pose some interoperability concerns for those who want to write JavaScript modules that work in both the browser and Node.js. The upcoming module loader specification is intended to help solve some of these module resolution interoperability concerns.
+
+#### Loading CommonJS Modules from JavaScript Modules
+
+Since there are millions of existing Node.js modules written in CommonJS format, the ability for JavaScript modules to load CommonJS modules is important. As previously mentioned, an `import` statement is capable of loading from a CommonJS module, although the semantics are a bit different due to how CommonJS defines modules. Suppose that the following CommonJS module is in the file `example.js`:
+
+```js
+module.exports.sum = function(num1, num2) {
+    return num1 + num2;
+};
+```
+
+This code exports a single function, `sum()` from a CommonJS module. To import just the `sum()` function from `example.js` in a JavaScript module, you can use the following:
+
+```js
+import { sum } from "./example";
+
+let result = sum(1, 2);
+```
+
+The `sum()` function is imported directly from `example.js` as if the imported module were a JavaScript module. Similarly, you can import everything defined on `module.exports` by importing `*`, such as:
+
+```js
+import * as example from "./example";
+
+let result = example.sum(1, 2);
+```
+
+Here, the local binding `example` is a new object that has references to every own property on `module.exports` object from `example.js` (note that `example` is not the same object as `module.exports`). That means you can access the `sum()` function as a method on `example`.
```

Comment:
new is misleading, the object is shared between imports,

if `a` imports `example` and `b` imports `example`, they would share the same namespace object.